### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by dodoburner**
In this PR my coding partner @shaqayq and I have:

Made sure that the N+1 problem is solved when fetching all posts and their comments for a user
Used Capybara to write integration tests for each view in the project according to the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md)

### CODE REVIEW

File: spec/views/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      expect(posts_counter[0]).to have_content('Number of posts: 0')
+      expect(posts_counter[1]).to have_content('Number of posts: 4')
+    end
+
+    it 'when I click on a user, I am redirected to that users show page' do
+      visit users_path
+      link = page.first('a')
+      link.click
+      expect(page).to have_current_path('/users/2')
+    end
```
Comment (TedLivist) [inline spec/views/user_spec.rb]: - [ ] I think these two are among the three tests that are failing. Kindly fix them.

  eyur

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @dodoburner and @shaqayq,

Good job so far!
✔️ Good documentation
✔️ Elimination of N+1 query problem
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/user_spec.rb (hunk 1 above)
Comment (dodoburner) [reply spec/views/user_spec.rb]: @TedLivist Everything is passing on my end, I don't know what I can do to make them pass on your end.

File: spec/views/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    it 'shows the number of posts each user has written' do
+      visit users_path
+      posts_counter = page.all('.posts-counter')
+      expect(posts_counter[0]).to have_content('Number of posts: 0')
+      expect(posts_counter[1]).to have_content('Number of posts: 4')
+    end
+
+    it 'when I click on a user, I am redirected to that users show page' do
```
Comment (codepantha) [inline spec/views/user_spec.rb]: - [ ] Just like the previous reviewer opined, these 3 tests are failing. Kindly see the attached image below try to fix them 👍 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi team👋,

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.

## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @dodoburner 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Proper use of Gitflow 👍 
* Good looking readme file 👍 
## Optional suggestions
*

>  I strongly suggest you update your setup section of your readme file on how to successfully run the tests and project as currently the tests are still failing like the other CRs pointed out. As debugging this issue is out of scope for this project it is approved. However we recommend an additional investigation in other to resolve this issue.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by dodoburner** [comment]
@Davidosky007 Weird because when I updated the tests, I asked a friend to check in his environment and we were on zoom together when he ran the tests and they were all passing, at least the spec/view files since that was the requirement of the project, I know that some of the previous ones are broken now but the views were passing.

**Comment by Davidosky007** [comment]
> @Davidosky007 Weird because when I updated the tests, I asked a friend to check in his environment and we were on zoom together when he ran the tests and they were all passing, at least the spec/view files since that was the requirement of the project, I know that some of the previous ones are broken now but the views were passing.

Well done @dodoburner since you have tested it on another environment, also update your setup section of your readme file too with the steps and commands you took

**Comment by dodoburner** [comment]
@Davidosky007 Thanks, I appreciate the input.

**Pull request merged**